### DISCUSSION TITLE
libusb: device or resource busy [code -6]

### DISCUSSION THREAD

**Discussion by GeoffAbbott** [danielpaulus/quicktime_video_hack#143] [Q&A] [answered]
I'm trying to get this running on a raspberry pi 4,bullsye, but I can't get past this issue when trying to retrieve the Config from the usbDevice. Anybody else run across this issue? Thanks

**Comment by Karazum** [comment, marked as answer]
I have run into that issue, but on linux.
Try building the latest usbmuxd from the source
https://github.com/libimobiledevice/usbmuxd

**Reply by GeoffAbbott** [reply]
Yes this worked for me! Also when stuck "waiting for ping..." it works the first time after activating but I have to un plug the device and plug it back in to get it to communicate again. I'm getting an error "device busy [code -6]" when I'm attempting to close the session so I'm assuming that is the issue.

**Reply by Karazum** [reply]
You can work around that one by using usb reset 
https://github.com/Karazum/usbreset/blob/master/usbreset.c

Basically, the flow for continues usage is 
```
if (qvh active?) {
    usbreset /dev/bus/usb/xxx/yyy
}
qvh activate
qvh gstreamer
```

but after some time, event that wont do it, nor will unplugging and replugging in. Something is wrong with the device itself.
Currently, before doing usb reset, I also kill mediaserverd and mediaremoted processes and so far it works.
